src/jit/emitxarch.cpp

```diff
@@ -2971,12 +2987,20 @@ regNumber emitter::emitInsBinary(instruction ins, emitAttr attr, GenTree* dst, G
         varNum = lclVar->AsLclVarCommon()->GetLclNum();
         offset = 0;
     }
+    else if (tmpDsc != nullptr)
+    {
+        varNum = tmpDsc->tdTempNum();
+        offset = 0;
+    }
 
-    if (varNum != BAD_VAR_NUM)
+    // spill temp numbers are negative.
+    if (varNum != BAD_VAR_NUM ||
+        tmpDsc != nullptr)
     {
```

Comment:
Spill temp numbers are assigned using Compiler::tmpCount, which is initialized to zero. Each time to allocate a temp, it is incremented and its sign is negated to use as its varNum (see tmpPreAllocateTemps).  As a result, the very first spill temp will have varNum = -1 which also happens to be BAD_VAR_NUM.  For that reason, the second condition is needed.

I have updated comment to be more clear of this reason.